src/cheri/cheri-pte-ext.adoc

```diff
@@ -79,45 +83,63 @@ as usual, and capability reads are controlled by the CRG bit.
 NOTE: The {ctag} bit of the stored capability is checked _after_ it is potentially
 cleared due to lack of <<c_perm>>.
 
-If the CW bit is clear then:
+
+If the CW bit is clear, the "no capability state", then the existing rules from <<section_priv_cheri_vmem>> are followed:
 
 * When a capability load or AMO instruction is executed, the implementation
  clears the {ctag} bit of the capability read from the virtual page.
-* When CRG is clear, the "no capability state", a {cheri_excep_name_pte_st}
- is raised when a capability store or AMO instruction is executed and the {ctag} bit
- of the capability being written is set.
-* When CRG is set, the "pre-CW state", two schemes are permitted (also see <<section_hardware_pte_updates>>):
-** The same behavior as when CRG is clear, allowing software interpretation
-of this state.
-** When a capability store or AMO instruction is executed
-and the {ctag} bit of the capability being written is set, the
-implementation sets the CW bit and assigns the CRG bit equal to <<sstatusreg_pte,sstatus>>.UCRG.
-+
-The PTE update must be
-atomic with respect to other accesses to the PTE, and must atomically check
-that the PTE is valid and grants sufficient permissions. Updates to the CW bit
-and CRG bit
-must be exact (i.e., not speculative), and observed in program order by the
-local hart. Furthermore, the PTE update must appear in the global memory order
-no later than the explicit memory access, or any subsequent explicit memory
-access to that virtual page by the local hart.  The ordering on loads and
-stores provided by FENCE instructions and the acquire/release bits on atomic
-instructions also orders the PTE updates associated with those loads and
-stores as observed by remote harts.
-+
-The PTE update is not required to be atomic with respect to the explicit memory
-access that caused the update, and the sequence is interruptible. However, the
-hart must not perform explicit memory access before the PTE update is globally
-visible.
+* When a capability store or AMO instruction is executed and the {ctag} bit
+ of the capability being written is set, the implementation raises a
+ {cheri_excep_name_pte_st}.
+
+When the CW bit is set, the "capability state", then the behaviour of
+capability loads is controlled by CRG and the behaviour of capability stores
+is controlled by CD.
```

Comment:
Gating CRG behavior on CW means we would be in trouble if ever we wanted PTEs that grant W and not CW for (aliased/shared) pages that hold caps (such as those that have a W and CW PTE elsewhere).  I don't know that we want such a thing, but it'd be good to (informatively) `[NOTE]` that this is a constraint on software behavior.

Reply:
Why is this a problem though? I  think that you can have a {W, ~CW} alias of a {W, CW} page, because in Svucrg CW gates both capability reads and writes, so you can't ever load a valid capability through the ~CW alias anyway. I think CW is a somewhat misleading name for the bit in this model, perhaps simply calling it `C` bit might be better and mirrors what we do in the capability HW permission bits with the "C-permission bit".